### ISSUE TITLE
proxmox_virtual_environment_file json unmarshalling type issue

### ISSUE THREAD

**Issue by Patricol** [bpg/terraform-provider-proxmox#41]
Error when refreshing state of proxmox_virtual_environment_file resources, both during creation and when planning with resources created previously.

`Error: Failed to decode HTTP GET response (path: nodes/NODE_NAME/storage/local/content) - Reason: json: cannot unmarshal string into Go struct field VirtualEnvironmentDatastoreFileListResponseData.data.size of type int`

Example resource:
```
resource "proxmox_virtual_environment_file" "iso-netboot" {
  content_type = "iso"
  datastore_id = "local"
  node_name = "NODE_NAME"
  source_file {
    path = "https://boot.netboot.xyz/ipxe/netboot.xyz-efi.iso"
  }
}
```

Running Proxmox 7.1 and version 0.5.0 of this provider. The old danitso 0.4.4 also encounters this bug.

I may work on fixing this myself.

P.S. thank you so much for your work on updating this provider! I switched from Telmate to danitso a year ago, and am hoping to avoid another switch.

**Comment by Patricol** [comment]
I may be misreading this, but I think the [api docs](https://pve.proxmox.com/pve-docs/api-viewer/index.html#/nodes/{node}/storage/{storage}/content/{volume}) say it is indeed supposed to be an int

**Comment by Patricol** [comment]
Actually running the GET, some of the sizes are ints and some are strings.

Pulling two out:

```
{
  "data": [{
    "ctime": 1645075082,
    "content": "images",
    "size": "4194304",
    "parent": null,
    "vmid": 100,
    "format": "qcow2",
    "used": "25088",
    "volid": "local:100/vm-100-cloudinit.qcow2"
  }, {
    "volid": "local:iso/netboot.xyz-efi.iso",
    "format": "iso",
    "content": "iso",
    "size": 1835008,
    "ctime": 1618813612
  }]
}
```
looks like all the string sizes are from cloudinit disks.

**Comment by Patricol** [comment]
Opened an [upstream issue](https://bugzilla.proxmox.com/show_bug.cgi?id=3894).

**Comment by bpg** [comment]
@Patricol thanks for looking into this! PRs are always welcome! :)

I see there is some activity on the upstream ticket, which is great. Hopefully will be fixed upstream and we don't need to do any patching in the provider.

**Comment by Patricol** [comment]
`libpve-storage-perl (7.1-2)` released on April 6th; with `fix #3894: cast 'size' and 'used' to integer` in the changelog.

I've confirmed that update fixes the issue I was having.
